Kernel/Process.cpp

```diff
@@ -58,98 +58,89 @@ MutexProtected<OwnPtr<KString>>& hostname()
     return *s_hostname;
 }
 
-SpinlockProtected<Process::List, LockRank::None>& Process::all_instances()
+SpinlockProtected<Process::AllProcessesList, LockRank::None>& Process::all_instances()
 {
     return *s_all_instances;
 }
 
 ErrorOr<void> Process::for_each_in_same_jail(Function<ErrorOr<void>(Process&)> callback)
 {
-    ErrorOr<void> result {};
-    Process::all_instances().with([&](auto const& list) {
-        Process::current().jail().with([&](auto const& my_jail) {
-            for (auto& process : list) {
-                if (!my_jail) {
+    return Process::current().m_jail_process_list.with([&](auto const& list_ptr) -> ErrorOr<void> {
+        ErrorOr<void> result {};
+        if (list_ptr) {
+            list_ptr->attached_processes().with([&](auto const& list) {
```

Comment:
(iʼd go for `list_ptr` → `list` and `list` → `processes` here instead; same in other places)

Reply:
It is here to ensure people understand there is an indirection here. This is not the actual list and therefore feels awkward to treat it like so.